### PR TITLE
.align-center-middle as new alignment class

### PR DESCRIPTION
**Pull request by paxperscientiam**
Hello,

I've written a new class named `.align-center-middle` for the flex-grid. It does both vertical centering -- `.align-middle` -- and horizontal centering -- `.align-center`.

I made sure to leverage extant mixins.

EDIT: I've changed the language of the OP to reflect the ensuing discussion.
EDIT2: This new class also causes immediate descendants to "gravitate" towards the parent's center (see `.align-contents: center` on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/align-content)).

### CODE REVIEW

**Comment by kball** [comment]
Oooh I like it. @rafibomb what do you think?

**Comment by IamManchanda** [comment]
Fallowing the convention that foundation uses .... I will go for `align-center-middle` to not complicate it much

File: scss/components/_flex.scss [hunk 1]
```diff
@@ -17,6 +17,13 @@
     }
   }
 
+  .align-central {
+    align-content: center;
+    @include flex-align($x: 'center');
+    @include flex-align($y: 'middle');
```
Comment (IamManchanda) [inline scss/components/_flex.scss]: Please rename to `align-center-middle`

**Review by IamManchanda — CHANGES_REQUESTED** [review]
This below will help with no complications and better convention .... Center and central are so similar that it get's hard to understand the use case of which is what

Poke @kball what do you think 

--------------------------------------------------------------------------------------------------------------

@paxperscientiam after its gets finalized, 
It will be great if you can provide a quick doc update too!
All in all, a very good addition to foundation.... thanks!!

File: scss/components/_flex.scss (hunk 1 above)
Comment (IamManchanda) [inline scss/components/_flex.scss]: Also, Please change the internal scss to this below 

```scss
.align-center-middle {
    align-content: center;
    @include flex-align($x: center, $y: middle);
 }
```
Ref. => http://foundation.zurb.com/sites/docs/flexbox.html#helper-mixins

**Comment by kball** [comment]
I tend to concur with @IamManchanda `.align-center-middle` is more explicit and less likely to confuse

**Comment by IamManchanda** [comment]
> I tend to concur with @IamManchanda `.align-center-middle` is more explicit and less likely to confuse

So @paxperscientiam, Review no.3 
Rename the title of this PR too apart from the commit :smiling_imp:

**Comment by paxperscientiam** [comment]
@IamManchanda @kball Thanks for the input -- I'm on it!

**Comment by IamManchanda** [comment]
=> **/docs/pages/flexbox.md**

Add this after (next in order) to [horizontal](http://foundation.zurb.com/sites/docs/flexbox.html#horizontal-alignment) and [vertical](http://foundation.zurb.com/sites/docs/flexbox.html#vertical-alignment) alignment and just above [Helper classes](http://foundation.zurb.com/sites/docs/flexbox.html#vanilla-flexbox-helper-classes) 

& I am unsure about the title @paxperscientiam @kball  .....

**Comment by rafibomb** [comment]
I love it! `.align-center-middle` is pretty good.

**Comment by paxperscientiam** [comment]
After @IamManchanda told me to RTFM (lol), it dawned on me that this class should probably be complemented with a "-self-" class.

EG:
````css
.align-self-center-middle {
 align-self: center;
 margin: 0 auto;
}

I'll whip it up, but please disabuse me of this notion if it's actually needless.

**Comment by IamManchanda** [comment]
@paxperscientiam `align-self-middle` is almost solving the same thing .....
Can you explain what extra are you doing/providing with that dreaded `margin: auto` ?

**Comment by paxperscientiam** [comment]
@IamManchanda Given a maximally agnostic parent (align-content default is stretch) I don't believe a flex item can be centered without `margin: auto`.

In either case, I should keep this PR atomic. Rolling back...

**Comment by paxperscientiam** [comment]
OK, updated `flexbox.md`.

File: docs/pages/flexbox.md [hunk 2]
```diff
@@ -164,6 +164,23 @@ To align an individual child, use the below classes. They use the same alignment
...
+Central alignment can be applied to a flex parent&mdash;which will centrally align all children automatically.
+
+To set parent alignment, use this class:
+
+- `.align-center-middle`
+
+```html_example
+<div class="row align-center-middle">
+  <div class="columns">I am in the center-middle&mdash; that is, I am centrally located!</div>
+  <div class="columns">As above, though I share the center-middle with my siblings.
```
Comment (IamManchanda) [inline docs/pages/flexbox.md]: You forgot to close the div... this is breaking the page layout

**Review by IamManchanda — CHANGES_REQUESTED** [review]

**Comment by IamManchanda** [comment]
@paxperscientiam Would request you to always please check locally before pushing changes. Your `flexbox.md` file had some errors both from written docs and coding perspective 

and sorry but as it was hard for me, to review the corrections and suggest changes line by line .... 
I have **FTFY** .... Please find the file below and update it with your current `flexbox.md` and then push the file again. Thanks .... Harry!

=> https://gist.github.com/IamManchanda/ab4b6d6d84152caa240a281f51a06c0a

**Comment by IamManchanda** [comment]
Expecting a commit @paxperscientiam ?

**Comment by paxperscientiam** [comment]
Sorry, I've been swamped. Will try to push today.

**Comment by IamManchanda** [comment]
No issue ..... @paxperscientiam ..... actually i wanna do a Pull request that includes this portion but i can't do that unless this is merged ... So that's why i am awaiting this merge !

**Comment by IamManchanda** [comment]
@paxperscientiam you updated old file again ..... did you replace the file with my given file??
=> https://gist.github.com/IamManchanda/ab4b6d6d84152caa240a281f51a06c0a

Are you finding any difficulty in the same???

**Comment by paxperscientiam** [comment]
@IamManchanda I hadn't noticed your file. Thanks. Let me put that in there. brb

**Comment by IamManchanda** [comment]
@paxperscientiam As you mentioned me on commit, for your personal benefit 

> Please use semantic names for commit/issues/PRs. It should describe the changes you made (with a complete description for commits), and not what you personally done.
> 
> :white_check_mark: Good:
> 
> 1. Renamed `align-central` to `align-center-middle`
> 2. Moved `align-center-middle` code to a mixin
> 3. Fix #xxx - Correct wrong variable name in `align-center-middle`
> 
> :x: Bad:
> 
> 1. Update Central alignement
> 2. **Apply changes asked by @someone**
> 3. I fixed #xxx, Yay!!
> 4. Here some changes, btw beautiful weather today, isn't it

**Review by IamManchanda — APPROVED** [review]
LGTM 
@kball for final review!

**Comment by paxperscientiam** [comment]
@IamManchanda 

> > Please use semantic names for commit/issues/PRs.

Is that lesson just for future application or do you want me to update the commit message?

**Comment by IamManchanda** [comment]
> > Please use semantic names for commit/issues/PRs.
> 
> Is that lesson just for future application or do you want me to update the commit message?

Keep it like this only for the time being .... let kevin sir decide on it!

**Comment by IamManchanda** [comment]
Ping @kball ..... awaiting final review ??

**Comment by kball** [comment]
Great work, merged!

**Pull request merged**